### PR TITLE
Add blog about Tasty Reader

### PR DESCRIPTION
**Pull request by bishabosha**

### CODE REVIEW

File: blog/_posts/2020-10-12-scala-3-forward-compat.md [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+To illustrate this scenario, we will start with an empty directory and
+build an sbt project with Scala 2.13 that has two modules, `shared`,
+which has some common domain model data structures, and `app`,
+which uses those data structures.
+
+For this project, we will pick
+[sbt 1.4.0](https://github.com/sbt/sbt/releases/tag/v1.4.0),
+this allows you to mix projects of different Scala versions with very
+little extra effort, (thanks to Eugene Yokota).
```
[the remark this answers is not in the document]
Comment (bishabosha) [reply blog/_posts/2020-10-12-scala-3-forward-compat.md]: thanks

**Review by julienrf — APPROVED** [review]

**Comment by SethTisue** [comment]
I think the main body of the post is good.

I suggest that the intro aim at including all of the important information in brief, so that someone will get the essential points they need even if they don't go on and read the whole thing. Condense the entire blog post into a single page; begin with just that page; and then follow it with the more detailed version.

An example of an important piece of information that isn't revealed until quite late in the blog post is that only some Scala 3 features are consumable from Scala 2.

**Review by SethTisue — COMMENTED** [review]
I think the main body of the post is good. It is already publishable.

But, I do suggest that the intro aim at including all of the important information in brief, so that someone will get the essential points they need even if they don't go on and read the whole thing. Condense the entire blog post into a single page; begin with just that page; and then follow it with the more detailed version.

An example of an important piece of information that isn't revealed until quite late in the blog post is that only some Scala 3 features are consumable from Scala 2.

**Comment by SethTisue** [comment]
Oh, one more thing. 2.13.14 comes out in just a couple weeks. Up to you, but I think you might want to wait until 2.13.4 is actually out, then you won't need to include the information about using a 2.13.14 nightly.

**Comment by bishabosha** [comment]
I've added more at the start, theres more of a problem with the librarey dependencies section due to there being no libraries currently for Scala `3.0.0-M1`, perhaps that section should be redesigned in some way

**Comment by bishabosha** [comment]
added more to the overview section, and removed the mentions of snapshot scala versions

**Comment by SethTisue** [comment]
intro LGTM now :+1:

I wonder if you want to say a few words about the status of `-Ytasty-reader` and/or about the overall status of the feature in terms of experimental vs. non-experimental, is requiring the flag considered temporary, that kind of thing.

I was a bit surprised to see `sourcecode` among the dependencies. Isn't it a macro, so isn't it surprising that it would work?

**Comment by bishabosha** [comment]
> I wonder if you want to say a few words about the status of `-Ytasty-reader` and/or about the overall status of the feature in terms of experimental vs. non-experimental, is requiring the flag considered temporary, that kind of thing.
 
Thats a good point, I should update to mention this.

> I was a bit surprised to see `sourcecode` among the dependencies. Isn't it a macro, so isn't it surprising that it would work?

Yes perhaps I should use a fictional library - tasty reader is lazy enough to only error when the code directly requires something unsupported - so a test suite that requires `sourcecode.Position` will fail

**Comment by bishabosha** [comment]
I've changed the blog to use ScalaCheck now that it has released for Scala `3.0.0-M1`, and I've added a section to talk about the `Ytasty-reader` flag

**Comment by bishabosha** [comment]
updated the date for Friday the 13th

**Comment by SethTisue** [comment]
We've had to push 2.13.4 out a bit again, as a few issues/PRs are still in flight, and then we'll need to float an "RC" (a designated nightly) and run the community build. Current target date is Mon Nov 16

**Comment by bishabosha** [comment]
@SethTisue when is the proposed date now?

**Comment by SethTisue** [comment]
reverting until 2.13.4 has been published: https://github.com/scala/scala-lang/pull/1179

**Comment by SethTisue** [comment]
draft PR to re-add when the time comes: #1180

**Comment by SethTisue** [comment]
> when is the proposed date now?

@bishabosha 2.13.4-bin-74cffdd is our release candidate. if all goes well, we'll re-release it on 2.13.4 on Thursday. let us know if you'd like any changes to the tasty-reader parts of https://github.com/scala/scala-dev/blob/scala-dev/releases/2.13.4.md

**Pull request merged**